mapie/conformity_scores/utils.py

```diff
@@ -1,26 +1,96 @@
-import numpy as np
-from mapie._typing import NDArray
+from typing import Optional
 
+from .regression import BaseRegressionScore
+from .classification import BaseClassificationScore
+from .bounds import AbsoluteConformityScore
+from .sets import APS, LAC, Naive, RAPS, TopK
 
-def get_true_label_position(y_pred_proba: NDArray, y: NDArray) -> NDArray:
+
+def check_regression_conformity_score(
+    conformity_score: Optional[BaseRegressionScore],
+    sym: bool = True,
+) -> BaseRegressionScore:
     """
-    Return the sorted position of the true label in the
-    prediction
+    Check parameter ``conformity_score`` for regression task.
+
+    Raises
+    ------
+    ValueError
+        If parameters are not valid.
 
-    Parameters
-    ----------
-    y_pred_proba: NDArray of shape (n_samples, n_classes)
-        Model prediction.
+    Examples
+    --------
+    >>> from mapie.conformity_scores.utils import (
+    ...     check_regression_conformity_score
+    ... )
+    >>> try:
+    ...     check_regression_conformity_score(1)
+    ... except Exception as exception:
+    ...     print(exception)
+    ...
+    Invalid conformity_score argument.
+    Must be None or a ConformityScore instance.
+    """
+    if conformity_score is None:
+        return AbsoluteConformityScore(sym=sym)
+    elif isinstance(conformity_score, BaseRegressionScore):
+        return conformity_score
+    else:
+        raise ValueError(
+            "Invalid conformity_score argument.\n"
+            "Must be None or a ConformityScore instance."
+        )
 
-    y: NDArray of shape (n_samples)
-        Labels.
 
-    Returns
-    -------
-    NDArray of shape (n_samples, 1)
-        Position of the true label in the prediction.
+def check_classification_conformity_score(
+    conformity_score: Optional[BaseClassificationScore] = None,
+    method: Optional[str] = None,
+) -> BaseClassificationScore:
     """
-    index = np.argsort(np.fliplr(np.argsort(y_pred_proba, axis=1)))
-    position = np.take_along_axis(index, y.reshape(-1, 1), axis=1)
+    Check parameter ``conformity_score`` for classification task.
 
-    return position
+    Raises
+    ------
+    ValueError
+        If parameters are not valid.
+
+    Examples
+    --------
+    >>> from mapie.conformity_scores.utils import (
+    ...     check_classification_conformity_score
+    ... )
+    >>> try:
+    ...     check_classification_conformity_score(1)
+    ... except Exception as exception:
+    ...     print(exception)
+    ...
+    Invalid conformity_score argument.
+    Must be None or a ConformityScore instance.
+    """
+    allowed_methods = ['lac', 'naive', 'aps', 'raps', 'top_k']
+    deprecated_methods = ['score', 'cumulated_score']
+    if method is not None:
+        if method in ['score', 'lac']:
```

Comment:
add depreciation warning to say that the user should use the conformity_score argument instead of method

Reply:
maybe it should be added in the `check_depreciated` method of MapieClassifier

Reply:
I moved it in utils.py, but it is still possible to do that definitively.

Reply:
Yes it would be great to have a depreciation warning if method is not None

Reply:
Done 👍